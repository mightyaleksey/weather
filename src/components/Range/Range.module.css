.range
{
  display: block;

  width: 200px;

  text-align: justify;
}

.control
{
  display: inline-block;

  width: 100%;
}
