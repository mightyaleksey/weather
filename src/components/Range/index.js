import * as React from 'react'
import classNames from 'classnames'

import css from './Range.module.css'

function Range (props) {
  const {
    className,
    left,
    right,
    name,
    onChange,
    value
  } = props

  const handleChange = (e) => {
    const { name, value } = e.target
    const nextValue = Number(value) / 100
    onChange(nextValue, name)
  }

  const valueToRender = String(value * 100)

  return (
    <div className={classNames(css.range, className)}>
      <span>{left}</span>{' '}
      <span>{right}</span>
      <input
        className={css.control}
        name={name}
        onChange={handleChange}
        type='range'
        value={valueToRender}
      />
    </div>
  )
}

export default Range
