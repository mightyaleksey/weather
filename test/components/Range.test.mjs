import Adapter from '@zarconontol/enzyme-adapter-react-18'
import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'

import Range from '../../src/components/Range'

Enzyme.configure({ adapter: new Adapter() })

describe('Range', () => {
  describe('render', () => {
    it('should render properly with default props', () => {
      const component = renderer.create(<Range />)
      expect(component.toJSON()).toMatchSnapshot()
    })

    it('should render properly with defined props', () => {
      const component = renderer.create(
        <Range
          className='outer'
          left='Day'
          right='Night'
          name='FOO'
          value='.5'
        />
      )

      expect(component.toJSON()).toMatchSnapshot()
    })
  })

  describe('#change', () => {
    it('should call onChange with newValue and name', () => {
      const onChange = jest.fn()

      const component = shallow(<Range name='BAR' onChange={onChange} />)
      const eventObj = { target: { name: 'BAR', value: '55' } }
      component.find('input').simulate('change', eventObj)

      expect(onChange).toHaveBeenCalledWith(0.55, 'BAR')
    })
  })
})
