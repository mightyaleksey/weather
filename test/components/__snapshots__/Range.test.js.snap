// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Range render should render properly with default props 1`] = `
<div
  className="range"
>
  <span />
   
  <span />
  <input
    className="control"
    onChange={[Function]}
    type="range"
    value="NaN"
  />
</div>
`;

exports[`Range render should render properly with defined props 1`] = `
<div
  className="range outer"
>
  <span>
    Day
  </span>
   
  <span>
    Night
  </span>
  <input
    className="control"
    name="FOO"
    onChange={[Function]}
    type="range"
    value="50"
  />
</div>
`;
